internal/batches/ui/interval_writer.go

```diff
@@ -116,15 +116,17 @@ type prefixedWriter struct {
 	prefix     string
 }
 
-func (w *prefixedWriter) Write(p []byte) (int, error) {
-	var prefixedLines []byte
-	for _, line := range bytes.Split(p, []byte("\n")) {
-		prefixedLine := append([]byte(w.prefix), line...)
-		prefixedLine = append(prefixedLine, []byte("\n")...)
+var newLineByteSlice = []byte("\n")
 
-		prefixedLines = append(prefixedLines, prefixedLine...)
-	}
-	w.writes <- prefixedLines
+func (w *prefixedWriter) Write(p []byte) (int, error) {
+	var prefixedLine []byte
+	// We remove new lines appended to the log output so all of the stream can have
+	// only one new line separating them. This ensures the previous behavior for the log
+	// structure is preserved.
+	p = bytes.TrimRight(p, "\n")
+	prefixedLine = append([]byte(w.prefix), p...)
+	prefixedLine = append(prefixedLine, newLineByteSlice...)
+	w.writes <- prefixedLine
```

Comment:
This could be simplified

```suggestion
	// We remove new lines appended to the log output so all of the stream can have
	// only one new line separating them. This ensures the previous behavior for the log
	// structure is preserved.
	p = bytes.TrimRight(p, "\n")
	prefixedLine := append([]byte(w.prefix), p...)
	prefixedLine = append(prefixedLine, newLineByteSlice...)
	w.writes <- prefixedLine
```

Reply:
Maybe this gives a small perf-boost but non-blocking:

```suggestion
	prefixedLine := append([]byte(w.prefix), p...)
	if prefixedLine[len(prefixedLine)-1] != newLineByteSlice {
		prefixedLine = append(prefixedLine, newLineByteSlice...)
    }
	w.writes <- prefixedLine
```